app.rb

```diff
@@ -0,0 +1,76 @@
+require 'sinatra'
+require 'sinatra/flash'
+require 'pg'
+require './lib/peeps'
+require './lib/user'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+
+  get '/' do
+    erb :index
+  end
+
+  get '/user/new' do
+    erb :new_user
+  end
+
+  post '/user' do
+    user = User.create(email: params[:email], password: params[:password], name: params[:name], username: params[:username])
+    session[:user_id] = user.id
+    session[:username] = user.username
+    redirect ('/peeps')
+  end
+
+  get '/peeps' do
+    @user = User.find(id: session[:user_id]) if session[:user_id]
+    # @username = User.find(id: session[:user_id]).username
+    @peeps = Peeps.all
+    erb :show
+  end
+
+  get '/new_peeps' do
+    erb :new_peep
+  end
+
+  get '/peeps/new' do
+    @user = User.find(id: session[:user_id]) if session[:user_id]
+
+    if @user
+      erb :show
+    else
+      flash[:notice] = "You need to sign in/log in to post a peep!"
+      redirect('/peeps')
+    end
```

Comment:
Nice! I like the idea to have a flash message for trying to peep without logging in